packages/bpk-component-autosuggestV2/src/BpkAutosuggest.tsx

```diff
@@ -0,0 +1,406 @@
+          <div className={theme.sectionTitle}>
+            {renderSectionTitle?.(section)}
+          </div>
+          {renderSuggestions(getSectionSuggestions?.(section)!, index)}
+        </section>
+      ));
+
+    const showSuggestions = alwaysRenderSuggestions ? true : isOpen;
+
+    const renderList = () =>
+      multiSection
+        ? renderSections(suggestions)
+        : renderSuggestions(suggestions);
+
+    return (
+      <div
+        className={getClassName(
+          theme.container,
+          suggestions.length && theme.containerOpen,
+        )}
+      >
+        {isBanana ? (
+          <label
+            {...getLabelProps({ 'aria-label': ariaLabels.label })}
+            className={getClassName(
+              theme.label,
+              isDesktop && theme.desktopLabel,
+            )}
+          >
+            <div className={theme.inputTextWrapper}>
+              {renderBesideInput?.()}
+              <div className={theme.inputWrapper}>
+                <input
+                  {...getInputProps({
+                    className: theme.input,
+                    onKeyDown,
+                    ref: forwardedRef,
+                    type: INPUT_TYPES.text,
+                    onClick: onClickOrKeydown,
+                    ...inputProps,
+                  })}
+                  enterKeyHint={enterKeyHint}
+                />
+              </div>
+            </div>
+            {inputValue.length > 0 && showClear && (
+              <div tabIndex={-1}>
+                <button
+                  type="button"
+                  title={ariaLabels.clearButton}
+                  onClick={clearSuggestions}
+                  aria-label={ariaLabels.clearButton}
+                  className={theme.clearButton}
+                  data-testid="clear button"
+                >
+                  <ClearButtonIcon />
+                </button>
+              </div>
+            )}
+          </label>
+        ) : (
+          <BpkInput
+            {...getInputProps({
+              onKeyDown,
+              onFocus: handleInputFocus,
+              ref: forwardedRef,
+              type: INPUT_TYPES.text,
+              onClick: onClickOrKeydown,
+              ...inputProps,
+            })}
+          />
+        )}
+
+        <div
+          className={getClassName(
+            theme.suggestionsContainer,
+            isDesktop && theme.desktopSuggestionsContainer,
+            multiSection && theme.sectionContainer,
+            showSuggestions && theme.suggestionsContainerOpen,
+          )}
+        >
+          <ul
+            {...getMenuProps({ 'aria-label': ariaLabels.resultsList })}
+            className={getClassName(
+              theme.suggestionsList,
+              isDesktop && theme.desktopSuggestionsList,
+            )}
+          >
+            {showSuggestions && renderList()}
+          </ul>
+        </div>
+      </div>
+    );
+  },
+);
+
+BpkAutosuggest.defaultProps = {
```

Comment:
we've stopped using this in BPK as it will be deprecated for functional components, you can instead destructure the props and set the default values using arguments?

Reply:
We need this here for the consumers to override the theming [here](https://github.com/Skyscanner/backpack/pull/2854#discussion_r1210310436)

Reply:
I meant the defaultProps in functional components, but yes also the theming 🙂